make build failed.

According to the edgexfoundry official website steps , the following error occurs.

The step from => https://docs.edgexfoundry.org/Ch-GettingStartedSDK-Go.html#build-your-device-service

![image](https://user-images.githubusercontent.com/29615989/65756327-78601080-e147-11e9-9817-07571d60a1f3.png)

I have the same problem: 
github.com/edgexfoundry/device-sdk-go/internal/clients/init.go:202:58: cannot use endpoint (type *endpoint.Endpoint) as type clients.Endpointer in argument to metadata.NewAddressableClient:
	*endpoint.Endpoint does not implement clients.Endpointer (missing Fetch method)

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,7 @@
 module github.com/edgexfoundry/device-sdk-go
 
 require (
-	github.com/edgexfoundry/go-mod-core-contracts v0.1.16
+	github.com/edgexfoundry/go-mod-core-contracts v0.1.30
 	github.com/edgexfoundry/go-mod-registry v0.1.0
 	github.com/google/uuid v1.1.0
 	github.com/gorilla/context v0.0.0-20181012153548-51ce91d2eadd // indirect
@@ -12,3 +12,5 @@ require (
 	github.com/ugorji/go v1.1.4
 	gopkg.in/yaml.v2 v2.2.2
 )
+
+go 1.12

--- a/internal/endpoint/endpoint.go
+++ b/internal/endpoint/endpoint.go
@@ -25,11 +25,10 @@ type Endpoint struct {
 
 func (endpoint Endpoint) Monitor(params types.EndpointParams, ch chan string) {
 	for {
-		data, err := endpoint.RegistryClient.GetServiceEndpoint(params.ServiceKey)
+		url, err := endpoint.buildURL(params)
 		if err != nil {
 			fmt.Fprintln(os.Stdout, err.Error())
 		}
-		url := fmt.Sprintf("http://%s:%v%s", data.Host, data.Port, params.Path)
 		ch <- url
 
 		// After the first run, the client can be indicated initialized
@@ -39,5 +38,25 @@ func (endpoint Endpoint) Monitor(params types.EndpointParams, ch chan string) {
 		}
 
 		time.Sleep(time.Second * time.Duration(params.Interval))
+	}
+}
+
+func (e Endpoint) Fetch(params types.EndpointParams) string {
+	url, err := e.buildURL(params)
+	if err != nil {
+		fmt.Fprintln(os.Stdout, err.Error())
+	}
+	return url
+}
+
+func (e Endpoint) buildURL(params types.EndpointParams) (string, error) {
+	if e.RegistryClient != nil {
+		endpoint, err := (e.RegistryClient).GetServiceEndpoint(params.ServiceKey)
+		if err != nil {
+			return "", fmt.Errorf("unable to get Service endpoint for %s: %s", params.ServiceKey, err.Error())
+		}
+		return fmt.Sprintf("http://%s:%v%s", endpoint.Host, endpoint.Port, params.Path), nil
+	} else {
+		return "", fmt.Errorf("unable to get Service endpoint for %s: Registry client is nil", params.ServiceKey)
 	}
 }

--- a/service.go
+++ b/service.go
@@ -146,7 +146,7 @@ func selfRegister() error {
 	ds, err := common.DeviceServiceClient.DeviceServiceForName(common.ServiceName, ctx)
 
 	if err != nil {
-		if errsc, ok := err.(*types.ErrServiceClient); ok && (errsc.StatusCode == http.StatusNotFound) {
+		if errsc, ok := err.(types.ErrServiceClient); ok && (errsc.StatusCode == http.StatusNotFound) {
 			common.LoggingClient.Info(fmt.Sprintf("Device Service %s doesn't exist, creating a new one", ds.Name))
 			ds, err = createNewDeviceService()
 		} else {
@@ -202,7 +202,7 @@ func makeNewAddressable() (*contract.Addressable, error) {
 	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.New().String())
 	addr, err := common.AddressableClient.AddressableForName(common.ServiceName, ctx)
 	if err != nil {
-		if errsc, ok := err.(*types.ErrServiceClient); ok && (errsc.StatusCode == http.StatusNotFound) {
+		if errsc, ok := err.(types.ErrServiceClient); ok && (errsc.StatusCode == http.StatusNotFound) {
 			common.LoggingClient.Info(fmt.Sprintf("Addressable %s doesn't exist, creating a new one", common.ServiceName))
 			millis := time.Now().UnixNano() / int64(time.Millisecond)
 			addr = contract.Addressable{

